sysprep/instance_setup.ps1

```diff
@@ -245,6 +245,24 @@ function Create-GCEStartup {
   $folder.RegisterTaskDefinition('GCEStartup',$task,6,'System',$null,5) | Out-Null
 }
 
+function Optimize-PowershellAssemblies {
+  # NGEN powershell assembly, improves startup time of powershell by 10x
+  $old_path = $env:path
+  try {
+    $env:path = [Runtime.InteropServices.RuntimeEnvironment]::GetRuntimeDirectory()
+    [AppDomain]::CurrentDomain.GetAssemblies() | % {
+      if (! $_.location) {continue}
+      $Name = Split-Path $_.location -leaf
+      if ($Name.startswith("Microsoft.PowerShell.")) {
+        Write-Progress -Activity "Native Image Installation" -Status "$name"
```

Comment:
Write-Progress can cause issues when there is no console attached (as is the case here), if you want this stuff printed out you can use the Write-Log function